ckanext/switzerland/commands.py

```diff
@@ -231,3 +255,64 @@ def _print_harvest_jobs(self, jobs):
                           job.id,
                           job.created.strftime('%Y-%m-%d %H:%M:%S'),
                           job.status))
+
+    def shacl_validate(self, source=None):
+        """
+        command for the harvester job cleanup
+        :argument source: string (optional)
+        """
+        # checking arguments
+        data_dict = {}
+        print('\nCommand Shacl Validation:\n')
+        if len(self.args) >= 2:
+            source_id = unicode(self.args[1])
+        else:
+            print('- Aborting: Please provide a harvest source')
+            sys.exit(1)
+
+        if not self.options.shapefile:
+            print('- Aborting: Please provide a shapefile name')
+            sys.exit(1)
+
+        shapedir = ogdch_helpers.get_shacl_shapesdir_from_config()
+        shapefilepath = os.path.join(
+            shapedir, self.options.shapefile)
+        if not os.path.exists(shapefilepath):
+            print('Shacl shape file does not exist in path {}'
+                  .format(shapefilepath))
+            sys.exit(1)
+
+        # loading arguments into config
+        data_dict['harvest_source_id'] = source_id
+        data_dict['shapefile'] = self.options.shapefile
+
+        # setting up other filepathes
+        data_dict['shapefilepath'] = shapefilepath
+        data_dict['resultdir'] = \
+            ogdch_helpers.make_shacl_results_dir(source_id)
+        data_dict['shaclcommand'] = \
+            ogdch_helpers.get_shacl_command_from_config()
+        data_dict['datapath'] = ogdch_helpers.get_shacl_file_path(
+            data_dict['resultdir'], 'data', 'ttl')
+        data_dict['resultpath'] = ogdch_helpers.get_shacl_file_path(
+            data_dict['resultdir'], 'result', 'ttl')
+        data_dict['csvpath'] = ogdch_helpers.get_shacl_file_path(
+            data_dict['resultdir'], 'result', 'csv')
+
+        # set context
+        context = {'model': model,
+                   'session': model.Session,
+                   'ignore_auth': True}
+        admin_user = logic.get_action('get_site_user')(context, {})
+        context['user'] = admin_user['name']
+
+        # test authorization
+        try:
+            logic.check_access('harvest_sources_clear', context, data_dict)
```

Comment:
this doesn't seem necessary - we're not clearing the harvest sources here, are we?